library/LuaTypes.cpp

```diff
@@ -706,6 +707,16 @@ static type_identity *find_primitive_field(lua_State *state, int field, const ch
  */
 static int meta_primitive_index(lua_State *state)
 {
+    const char *attr = lua_tostring(state, -1);
+    if (attr == std::string("ref_target")) {
```

Comment:
since you're comparing two c strings here, you can just do `!strcmp(attr, "ref_target")` to avoid a string allocation

Reply:
I could indeed, but I was lazy and I expect to change this to not use string comparisons at all (like the bitfield-handling code I mentioned)